py_openda/py_openda/utils/py4j_utils.py

```diff
@@ -0,0 +1,206 @@
+
+    # Translate the input strings to java objects (File and String array)
+    j_input_dir = gateway.jvm.java.io.File(input_dir)
+    string_class = gateway.jvm.String
+    j_arguments = gateway.new_array(string_class, 1)
+    j_arguments[0] = config_xml
+
+    # Initialize ..
+    openda_configurable.initialize(j_input_dir, j_arguments)
+
+def py_list_to_j_array(py_x):
+    """
+    Create a java double array from python list.
+
+    :param py_x: python list of doubles.
+    :return: java array of doubles copy of py_x.
+    """
+    n_elements = len(py_x)
+    double_class = gateway.jvm.double
+    j_x = gateway.new_array(double_class, n_elements)
+    j_x[:] = py_x[:]
+
+    return j_x
+
+def j_array_to_py_list(j_x):
+    """
+    Create a python list from a java array.
+    :param j_x: java array.
+    :return: python list with values of j_x.
+    """
+
+    n_elements = len(j_x)
+    py_x = [None]*n_elements
+    py_x[:] = j_x[:]
+    return py_x
+
+def j_vector_array_to_np_array(j_vectors):
+    """
+    Create a numpy array from a java vector array.
+    Note that this function also works for different data types,
+    as long as the data type has the method getValues().
+
+    :param j_vectors: java vector array.
+    :return: numpy array with values of j_vectors.
+    """
+
+
+    vals = [j_array_to_py_list(vec.getValues()) for vec in j_vectors]
+    py_array = np.c_[[np.array(val) for val in vals]].transpose()
+    return py_array
+
+def np_array_to_j_array(np_array):
+    """
+    Create a java double array from a 1D numpy array.
+
+    :param np_array: 1D numpy array of doubles.
+    :return: java array of doubles.
+    """
+    np_array = np_array.squeeze().tolist()
+    j_array = py_list_to_j_array(np_array)
+    return j_array
+
+def np_array_to_j_vector(np_array):
+    """
+    Create a java Vector from a 1D numpy array.
+
+    :param np_array: 1D numpy array of doubles.
+    :return: java Vector.
+    """
+    j_array = np_array_to_j_array(np_array)
+    return gateway.jvm.org.openda.utils.Vector(j_array)
+
+def input_to_j_array(obj):
+    """
+    Converts a python list, a 1D numpy array, java Array or an OpenDA vector
+    to a java array.
+
+    :param obj: object of unknown type.
+    :return: obj as java array
+    """
+    if isinstance(obj, list):
+        obj = py_list_to_j_array(obj)
+    elif isinstance(obj, np.ndarray):
+        obj = np_array_to_j_array(obj)
+    elif isinstance(obj, JavaArray):
+        None
+    elif IVector_class.isInstance(obj):
+        obj = obj.getValues()
+    return obj
+
+def input_to_py_list(obj):
+    """
+    Converts a python list, a 1D numpy array, java Array or an OpenDA vector
+    to a java array.
+
+    :param obj: object of unknown type.
+    :return: obj as java array
+    """
+    if isinstance(obj, JavaArray):
+        obj = j_array_to_py_list(obj)
+    elif isinstance(obj, np.ndarray):
+        obj = obj.squeeze().tolist()
+    elif isinstance(obj, list):
+        None
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Statement seems to have no effect](https://app.codacy.com/gh/OpenDA-Association/OpenDA/pullRequest?prid=4902598)